bindings_node/src/streams.rs

```diff
@@ -0,0 +1,38 @@
+use std::sync::{
+  atomic::{AtomicBool, Ordering},
+  Arc, Mutex,
+};
+use tokio::sync::oneshot::Sender;
+
+use napi_derive::napi;
+
+#[napi]
+pub struct NapiStreamCloser {
+  close_fn: Arc<Mutex<Option<Sender<()>>>>,
+  is_closed_atomic: Arc<AtomicBool>,
```

Comment:
these types were taken from the FFI bindings. not sure if they should remain the same for Node.

Reply:
The only constraint is that these methods must be exposed to the platform SDK (Node), so that they know if the stream is closed and have the ability to close it: https://github.com/xmtp/libxmtp/blob/main/bindings_ffi/src/mls.rs#L875-L888

I suspect we need to keep the `Mutex` and `AtomicBool` for Rust concurrency purposes, but I'm unsure if the `Arc` can be removed or not

Reply:
(For clarity, it's fine to leave as-is, if it's currently working)